src/vos/vos_pool.c

```diff
@@ -2114,3 +2114,365 @@ vos_pool_feature_skip_dtx_resync(daos_handle_t poh)
 
 	return vos_pool->vp_pool_df->pd_compat_flags & VOS_POOL_COMPAT_FLAG_SKIP_DTX_RESYNC;
 }
+
+static int
+vos_tgt_preallocate(uuid_t uuid, daos_size_t scm_size, int tgt_id, const char *storage_path)
+{
+	char *newborns_path = NULL;
+	char *path          = NULL;
+	int   fd            = -1, rc;
+
+	/** create the newborns_path string */
+	D_ASPRINTF(newborns_path, "%s/NEWBORNS", storage_path);
```

Comment:
I do not get why did not simply provide the "%s/NEWBORNS" as `storage_path`. You already generated it once. No point in generating it N more times, is it?